src/urllib3/contrib/socks.py

```diff
@@ -66,23 +67,37 @@
 try:
     import ssl
 except ImportError:
-    ssl = None
+    ssl = None  # type: ignore
+
+try:
+    from typing import TypedDict  # >=3.8
+except ImportError:
+    from typing_extensions import TypedDict  # <=3.7
```

Comment:
Can we explicit write "Python 3.7"? It's much easier to grep when we'll be removing support for 3.7